src/tests/Microsoft.PowerFx.Core.Tests/TexlTests.cs

```diff
@@ -117,5 +148,2026 @@ public void TexlDateTableFunctions_Negative(string expression)
+                script,
+                TestUtils.DT(expectedType),
+                symbol);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsInt()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[A:n]")));
+
+            TestSimpleBindingSuccess(
+                "Int(1234)",
+                DType.Number);
+
+            TestSimpleBindingSuccess(
+                "Int(T)",
+                TestUtils.DT("*[A:n]"),
+                symbol);
+        }
+
+        [Theory]
+        [InlineData("Int(\"3\")", "n", null)]
+        [InlineData("Int(true)", "n", null)]
+        [InlineData("Int(T)", "*[Booleans:n]", "*[Booleans:b]")]
+        [InlineData("Int(T)", "*[Strings:n]", "*[Strings:s]")]
+        [InlineData("Int([true, false, true])", "*[Value:n]", null)]
+        [InlineData("Int([\"5\", \"6\"])", "*[Value:n]", null)]
+        public void TexlFunctionTypeSemanticsIntWithCoercion(string script, string expectedType, string typedGlobal)
+        {
+            var symbol = new SymbolTable();
+            
+            if (typedGlobal != null)
+            {
+                symbol.AddVariable("T", new TableType(TestUtils.DT(typedGlobal)));
+                TestSimpleBindingSuccess(
+                    script,
+                    TestUtils.DT(expectedType),
+                    symbol);
+            }
+            else
+            {
+                TestSimpleBindingSuccess(script, TestUtils.DT(expectedType));
+            }
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsTruncOneParam()
+        {
+            TestSimpleBindingSuccess(
+                "Trunc(1234)",
+                DType.Number);
+
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[A:n]")));
+            TestSimpleBindingSuccess(
+                "Trunc(T)",
+                TestUtils.DT("*[A:n]"),
+                symbol);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsTruncTwoParams()
+        {
+            var symbol = new SymbolTable();
+            symbol.AddVariable("T", new TableType(TestUtils.DT("*[digits:n]")));
+            symbol.AddVariable("X", new TableType(TestUtils.DT("*[Nnnuuummm:n]")));
+
+            TestSimpleBindingSuccess(
+                "Trunc(1234.567, 4)",
+                DType.Number);
+
+            TestSimpleBindingSuccess(
+                "Trunc(1234.567, T)",
+                TestUtils.DT("*[Result:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "Trunc(X, T)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+
+            TestSimpleBindingSuccess(
+                "Trunc(X, 4)",
+                TestUtils.DT("*[Nnnuuummm:n]"),
+                symbol);
+        }
+
+        [Fact]
+        public void TexlFunctionTypeSemanticsRound()
```

Comment:
>TexlFunctionTypeSemanticsRound [](http://example.com/codeflow?start=20&length=30)

We should have a separate test that validates that the output is *[Value:n] if the ConsistentOneColumnTableResult feature is enabled. Same for other tabular functions